src/parser/xasl_generation.c

```diff
@@ -25217,6 +25204,37 @@ pt_to_merge_insert_query (PARSER_CONTEXT * parser, PT_NODE * select_list, PT_MER
   return subq;
 
 error_exit:
+
+  if (subq)
+    {
+      parser_free_tree (parser, subq);
+    }
+
+  if (expr)
+    {
+      if (expr->info.expr.arg1)
+	{
+	  parser_free_tree (parser, expr->info.expr.arg1);
+	  corr_subq = NULL;
+	}
+      parser_free_tree (parser, expr);
+    }
+
+  if (corr_subq)
+    {
+      parser_free_tree (parser, corr_subq);
+    }
```

Comment:
How about below?
```suggestion
  if (corr_subq)
    {
      parser_free_tree (parser, corr_subq);
    }
   
  if (expr)
    {
      parser_free_tree (parser, expr);
    }
```

Reply:
This code sequence is right because the expr could be assigend to **corr_subq**.